presto-hive/src/test/java/com/facebook/presto/hive/util/TestHiveFileIterator.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.util;
+
+import com.facebook.presto.hive.HadoopDirectoryLister;
+import com.facebook.presto.hive.NamenodeStats;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static com.facebook.presto.hive.util.HiveFileIterator.NestedDirectoryPolicy.IGNORED;
+import static com.facebook.presto.hive.util.HiveFileIterator.NestedDirectoryPolicy.RECURSE;
+import static com.google.common.io.Files.createTempDir;
+import static org.testng.Assert.assertEquals;
+
+public class TestHiveFileIterator
+{
+    private String pathFilterString = "path_filter_test_file_";
+    private Configuration hadoopConf;
+    private Random random;
+    private HiveFileIterator.ListDirectoryOperation listDirectoryOperation;
+    private static final String RANDOM_FILE_NAME_SALT_STRING = "abcdefghijklmnopqrstuvwxyz";
+
+    @BeforeClass
+    private void setup()
+    {
+        random = new Random();
```

Comment:
`Random` is not thread safe, and the tests are ran in multiple threads. Please use `ThreadLocalRandom` instead (inline `ThreadLocalRandom.current()` everywhere)

Reply:
removed random usage